### PR TITLE
5.4 to 6.0 merge

### PR DESCRIPTION
**Pull request by arekpalinski**
@grisha-kotler please review changes of https://github.com/ravendb/ravendb/pull/18469 applied to 6.0

### CODE REVIEW

File: test/SlowTests/Issues/RavenDB_22195.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public RavenDB_22195(ITestOutputHelper output) : base(output)
         }
 
         [RavenTheory(RavenTestCategory.Replication)]
-        [RavenData(DatabaseMode = RavenDatabaseMode.All)]
+        [RavenData(DatabaseMode = RavenDatabaseMode.Single)]
```
Comment (arekpalinski) [inline test/SlowTests/Issues/RavenDB_22195.cs:20]: @shiranshalom I have adjusted the test since I believe it was no supposed to check sharded endpoint since `ShardedReplicationHandler` doesn't have `debug/replication/all-items endpoint`. If this needs to be added please handle that in a dedicated PR.

**Comment by grisha-kotler** [comment]
> @grisha-kotler please review changes of #18469 applied to 6.0

LGTM
